core/src/main/java/io/undertow/server/protocol/ajp/AjpReadListener.java

```diff
@@ -215,6 +215,17 @@ public void handleEvent(AjpServerResponseConduit channel) {
             //we need to set the write ready handler. This allows the response conduit to wrap it
             responseConduit.setWriteReadyHandler(writeReadyHandler);
 
+            // Check if the number of cookies sent exceeded the maximum
+            try {
+                httpServerExchange.getRequestCookies();
```

Comment:
This is not a good approach for performance reasons, as this involves creating a cookie map that may not be used (e.g. if the request is just serving a resource and does not care about cookies). 

Any sort of check for this would also have to be dealt with in the HTTP/2 code to keep it consistent.